saleor/warehouse/availability.py

```diff
@@ -22,7 +24,12 @@ def _get_available_quantity(stocks: StockQuerySet) -> int:
     return max(total_quantity - quantity_allocated, 0)
 
 
-def check_stock_quantity(variant: "ProductVariant", country_code: str, quantity: int):
+def check_stock_quantity(
+    variant: "ProductVariant",
+    country_code: str,
+    quantity: int,
+    user: Optional["User"] = None,
```

Comment:
Again a loose idea but maybe we could decouple this module from users and just operate on stocks and reservations - instead of passing the user, we could pass a list of reservations to take into account when checking stock quantity. We wouldn't have to pass the user a few levels down. But the current implementation is also reasonable so I won't argue about that :)